### ISSUE TITLE
`CCL_ERROR_MEMORY` when calling `get_density_kernel` in version >= 2.4

### ISSUE THREAD

**Issue by jfcrenshaw** [LSSTDESC/CCL#960]
When I try to calculate a density kernel using CCL version >= 2.4, I get an error that says `CCLError: Error CCL_ERROR_MEMORY: `. This doesn't happen when I use version 2.3.

Here is a working example:
```python
import numpy as np
import pyccl as ccl

# setup the cosmology
h = 0.6774
cosmo = ccl.Cosmology(
    Omega_c=0.119 / h**2,
    Omega_b=0.022 / h**2,
    h=h,
    A_s=2.4e-9,
    n_s=0.967,
)

# setup a redshift grid
z = np.linspace(0.0, 4.0, 200)

# create a toy n(z)
nz = np.exp(-((z - 2) ** 2) / (2 * 0.2**2))
nz /= np.trapz(nz, z)

# create the kernel
kernel = ccl.get_density_kernel(cosmo, (z, nz))

```

and here is the full error message:
```python
---------------------------------------------------------------------------
CCLError                                  Traceback (most recent call last)
/Users/jfcrenshaw/Documents/rubin-lya-forecast/src/scripts/calculate_correlations.py in <cell line: 22>()
     24 nz /= np.trapz(nz, z)
     26 # create the kernel
---> 27 kernel = ccl.get_density_kernel(cosmo, (z, nz))

File /opt/homebrew/Caskroom/mambaforge/base/envs/rubin-lya/lib/python3.9/site-packages/pyccl/tracers.py:64, in get_density_kernel(cosmo, dndz)
     47 """This convenience function returns the radial kernel for
     48 galaxy-clustering-like tracers. Given an unnormalized
     49 redshift distribution, it returns two arrays: chi, w(chi),
   (...)
     61         to unity.
     62 """
     63 z_n, n = _check_array_params(dndz, 'dndz')
---> 64 _check_background_spline_compatibility(cosmo, dndz[0])
     65 # this call inits the distance splines neded by the kernel functions
     66 chi = comoving_radial_distance(cosmo, 1./(1.+z_n))

File /opt/homebrew/Caskroom/mambaforge/base/envs/rubin-lya/lib/python3.9/site-packages/pyccl/tracers.py:36, in _check_background_spline_compatibility(cosmo, z)
     32 def _check_background_spline_compatibility(cosmo, z):
     33     """Check that a redshift array lies within the support of the
     34     CCL background splines.
     35     """
---> 36     a_bg, _ = _get_spline1d_arrays(cosmo.cosmo.data.chi)
...
---> 48     raise CCLError("Error %s: %s" % (error_types[status], msg))
     50 # Check for unknown error
     51 if status != 0:

CCLError: Error CCL_ERROR_MEMORY: 
```

**Comment by damonge** [comment]
@jfcrenshaw I bet this was caused by this PR: https://github.com/LSSTDESC/CCL/pull/926

To test it, can you run the same code but adding
```
cosmo.compute_distances()
cosmo.compute_growth()
```
just after creating the cosmology?

**Comment by jfcrenshaw** [comment]
Adding `cosmo.compute_distances()` after creating the cosmology fixed the problem. Thanks!

**Comment by damonge** [comment]
OK, thanks! I've fixed it in #961, but until it makes it to a release you can use this fix.
